workflow-ui/internal-testing-android/src/main/java/com/squareup/workflow1/ui/internal/test/IdlingDispatcher.kt

```diff
@@ -0,0 +1,64 @@
+package com.squareup.workflow1.ui.internal.test
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.IdlingResource
+import androidx.test.espresso.idling.CountingIdlingResource
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Runnable
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Adapted from Dispatch
+ * https://github.com/RBusarow/Dispatch/blob/main/dispatch-android-espresso/src/main/java/dispatch/android/espresso/IdlingDispatcher.kt
+ *
+ * [IdlingResource] helper for coroutines.  This class simply wraps a delegate [CoroutineDispatcher]
+ * and keeps a running count of all coroutines it creates, decrementing the count when they complete.
+ *
+ * If a coroutine dispatches to another dispatcher, this dispatcher will remain non-idle until the
+ * outer coroutine has completed, or until the coroutines reach an *inactive* `suspend` point.  In
+ * other words, if the tracked coroutine is suspended because it's waiting on a blocking I/O
+ * operation, then the dispatcher will not be idle.  If the tracked coroutine is suspended because
+ * it's waiting on something like a timer or user input, then the dispatcher will transition to
+ * idle.
+ */
+public class IdlingDispatcher(
+  private val delegate: CoroutineDispatcher
+) : CoroutineDispatcher() {
+
+  /**
+   * The [CountingIdlingResource] which is responsible for [Espresso] functionality.
+   */
+  public val counter: CountingIdlingResource = CountingIdlingResource("IdlingResource for $this")
```

Comment:
Where are we actually bringing this to Espresso's attention?

Reply:
In the rule.

Reply:
Need a :derp: emoji. Oh, I guess we have one 🤦